jwst/ramp_fitting/ramp_fit.py

```diff
@@ -155,21 +153,35 @@ def ols_ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
     nreads, npix, imshape, cubeshape, n_int, instrume, frame_time, ngroups = \
         utils.get_dataset_info(model)
 
+    # Save original shapes for writing to log file, as these change for MIRI
+    orig_nreads = nreads
+    orig_cubeshape = cubeshape 
+        
+    # For MIRI datasets having >1 reads, resize the input model arrays to
+    # exclude the final group. 
+    if (instrume == 'MIRI' and nreads > 1):   
+        model.data = model.data[:,:-1,:,:]
+        model.err = model.err[:,:-1,:,:]
+        model.groupdq = model.groupdq[:,:-1,:,:]
+        nreads -= 1
+        ngroups -= 1
+        cubeshape = (nreads,)+imshape
+
```

Comment:
This approach of simply removing the data corresponding to the last group/frame for all integrations certainly accomplishes what's intended, but it also then hardwires the elimination of the last group into the code. The lastframe step is going to be modified to flag all of the data in the last group as bad, which should or could then trigger routines within ramp_fit to not use it, based on those DQ flags, rather than simply excluding it from the routines all together. The MIRI team has plans (or at least hopes) of eventually coming up with a suitable correction for the data in the last group/frame, at which point the lastframe step will be modified again and the data will no longer be flagged. So if ramp_fit just keys off of the flags, it would automatically do the right thing when/if that change occurs in the future. This approach, however, excludes those data without any knowledge of whether they're useable or not. So I would suggest trying to determine a way to have ramp_fit exclude the last group/frame data based only on the DQ flags that are set for them.